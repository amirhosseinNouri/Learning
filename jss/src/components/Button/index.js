import React from 'react';
import cx from '../../utils/cx';
import useStyle from './style';
import { useTheme } from 'react-jss';

export default function Button(props) {
  const theme = useTheme();
  const classes = useStyle({ ...props, theme });
  return (
    <button className={classes.button}>
      <span className={classes.label}>Click me</span>
    </button>
  );
}

Button.defaultProps = {
  spacing: 20,
  fontWeight: 'bold',
  labelColor: 'red',
};
