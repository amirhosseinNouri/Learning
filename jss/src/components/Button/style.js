import { createUseStyles, useTheme } from 'react-jss';

const useStyle = createUseStyles((theme) => ({
  button: {
    background: theme.colorPrimary,
  },
  label: {
    fontWeight: 'bold',
  },
}));

export default useStyle;
