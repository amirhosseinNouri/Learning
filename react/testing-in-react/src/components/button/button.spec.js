import React from 'react';
import Button from './button';
import renderer, { create, act } from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';

// test('Works', () => {
//   expect(true).toBe(true);
// });

// test('Renders the text', () => {
//   const text = 'text';
//   const component = renderer.create(<Button text={text} />);
//   expect(component.root.props.text).toBe(text);
// });

test('shit test', () => {
  let root;
  act(() => {
    root = create(<Button text={'text1'} />);
  });

  // make assertions on root
  expect(root.toJSON()).toMatchSnapshot();

  // update with some different props
  act(() => {
    root.update(<Button text={'text2'} />);
  });

  // make assertions on root
  expect(root.toJSON()).toMatchSnapshot();
});
