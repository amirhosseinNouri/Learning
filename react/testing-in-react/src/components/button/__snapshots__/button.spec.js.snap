// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `<div />`;

exports[` 2`] = `<div />`;

exports[`shit test 1`] = `<div />`;

exports[`shit test 2`] = `<div />`;
